### PR TITLE
 Download taxid descriptions

### PR DESCRIPTION
**Pull request by yunfangjuan**
download with 100 batch and 16 threads.

### CODE REVIEW

**Comment by yunfangjuan** [comment]
Ready for review~

File: idseq_dag/steps/fetch_tax_info.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+from idseq_dag.engine.pipeline_step import PipelineStep
+
+
+class PipelineStepFetchTaxInfo(PipelineStep):
+    '''
+        fetch tax info based on a list
+    '''
+    def run(self):
+        '''
+            1. fetch the taxid -> wikipedia link mappin
```
Comment (jshoe) [inline idseq_dag/steps/fetch_tax_info.py]: nit s/mappin/mapping

File: idseq_dag/steps/fetch_tax_info.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        # output dummay for actual wiki content for now
+        taxid2wikicontent = {}
+        self.fetching_wiki_content(taxid2wikidict, taxid2wikicontent)
+
+        with open(taxid2desc, 'w') as desc_outputf:
+            json.dump(taxid2wikicontent, desc_outputf)
+
+    @staticmethod
+    def fetching_wiki_content(taxid2wikidict, taxid2wikicontent):
```
Comment (jshoe) [inline idseq_dag/steps/fetch_tax_info.py]: nit maybe "fetch_wiki_content"?

File: idseq_dag/steps/fetch_tax_info.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        threads = []
+        semaphore = threading.Semaphore(64) # 4x the threads
```
Comment (jshoe) [inline idseq_dag/steps/fetch_tax_info.py]: maybe you could also use the additional_attributes["threads"] in this function?

File: idseq_dag/steps/fetch_tax_info.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                t = threading.Thread(
+                    target=PipelineStepFetchTaxInfo.
+                    get_wiki_content,
+                    args=[taxid, pageid, taxid2wikicontent, mutex, semaphore]
+                    )
+                t.start()
+                threads.append(t)
+        for t in threads:
+            t.join()
+    @staticmethod
```
Comment (jshoe) [inline idseq_dag/steps/fetch_tax_info.py]: nit for missing line break between these methods

File: idseq_dag/steps/fetch_tax_info.py (hunk 2 above)
Comment (yunfangjuan) [reply idseq_dag/steps/fetch_tax_info.py]: yep.

File: idseq_dag/steps/fetch_tax_info.py (hunk 3 above)
Comment (yunfangjuan) [reply idseq_dag/steps/fetch_tax_info.py]: yep.

**Comment by yunfangjuan** [comment]
you don't need an account for Entrez. just need to put an email there.

**Comment by jshoe** [comment]
@yunfangjuan you could change to accounts@idseq.net if you like

**Pull request merged**